pkg/network/ebpf/c/protocols/postgres-helpers.h

```diff
@@ -0,0 +1,45 @@
+#ifndef __POSTGRES_HELPERS_H
+#define __POSTGRES_HELPERS_H
+
+#include "postgres-defs.h"
+#include "sql-helpers.h"
+
+// is_postgres_connect checks if the buffer is a Postgres startup message.
+static __always_inline bool is_postgres_connect(const char *buf, __u32 buf_size) {
+    CHECK_PRELIMINARY_BUFFER_CONDITIONS(buf, buf_size, POSTGRES_STARTUP_MIN_LEN);
+
+    struct pg_startup_header *hdr = (struct pg_startup_header *)buf;
+
+    if (bpf_ntohl(hdr->version) != PG_STARTUP_VERSION) {
+        return false;
+    }
+
+    // Check if we can find the user param. Postgres uses C-style strings, so
+    // we also check for the terminating null byte.
+    return !bpf_memcmp(buf + sizeof(*hdr), PG_STARTUP_USER_PARAM, sizeof(PG_STARTUP_USER_PARAM));
+}
+
+// is_postgres_query checks if the buffer is a regular Postgres message.
+static __always_inline bool is_postgres_query(const char *buf, __u32 buf_size) {
+    CHECK_PRELIMINARY_BUFFER_CONDITIONS(buf, buf_size, sizeof(struct pg_message_header));
+
+    struct pg_message_header *hdr = (struct pg_message_header *)buf;
+
+    // We only classify queries for now
+    if (hdr->message_tag != POSTGRES_QUERY_MAGIC_BYTE && hdr->message_tag != POSTGRES_COMMAND_COMPLETE_MAGIC_BYTE) {
+        return false;
+    }
+
+    __u32 message_len = bpf_ntohl(hdr->message_len);
+    if (message_len < POSTGRES_MIN_PAYLOAD_LEN || message_len > POSTGRES_MAX_PAYLOAD_LEN) {
+        return false;
+    }
+
+    return is_sql_command(buf + sizeof(*hdr), buf_size - sizeof(*hdr));
+}
```

Comment:
I suspect there is an underlying bug here, postgres can chain commands
so for example, the command-complete you are looking for, can come second in line (after row description) in an example of select.

you should be more robust, and possible test more postgres methods

hint, disable classification by query `POSTGRES_QUERY_MAGIC_BYTE`, and run `postgres - select` test while wireshark is open.
you'll see the response has row description in the beginning of the payload